apis/rust/node/src/lib.rs

```diff
@@ -88,9 +90,10 @@ impl DoraNode {
                 .send_prepared_message(sample)
                 .wrap_err_with(|| format!("failed to send data for output {output_id}"))?;
         } else {
-            data(&mut []);
+            let mut buffer = vec![0; data_len];
```

Comment:
```suggestion
            let mut buffer = Vec::with_capacity(data_len);
```

Would be slightly faster, as there it does not allocate zeroes. But it's mainly because i saw it.

Reply:
`with_capacity` does not work here because then the `Vec`'s length would stay remain 0. The `data` closure expects a `&mut [u8]` slice with the requested length. 

We could use `MaybeUninit`, but then we would need to trust the user-provided code to properly initialize the data. Otherwise undefined behavior would occur when we access it. Given that the allocation size is quite small, I don't think that this is worth it.

Reply:
Ah true, because it needs the vec's length. Does that mean that if the user input the wrong `data_len`, there is going to be an Error?

Reply:
Only in the user code. The user requests a slice of a certain size and the daemon prepares that slice. The user code cannot change the size afterwards, so nothing can go wrong in the daemon. But the user code can of course trigger an out of bounds panic if they access an invalid slice index.